src/Illuminate/Notifications/Messages/SlackAttachmentAction.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+namespace Illuminate\Notifications\Messages;
+
+use Closure;
+
+class SlackAttachmentAction
+{
+    /**
+     * The name field of the attachment action.
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * The content of the attachment action.
+     *
+     * @var string
+     */
+    protected $content;
+
+    /**
+     * The type of the attachment action.
+     *
+     * @var string
+     */
+    protected $type = 'button';
+
+    /**
+     * The value of the attachment action.
+     *
+     * @var string
+     */
+    protected $value;
+
+    /**
+     * The action's confirmation fields.
+     *
+     * @var array
+     */
+    protected $confirmationFields;
+
+    /**
+     * The style of the attachment action.
+     *
+     * @var string
+     */
+    protected $style = 'default';
+
+    /**
+     * The options of the attachment action.
+     *
+     * @var array
+     */
+    protected $options;
+
+    /**
+     * The option groups of the attachment action.
+     *
+     * @var array
+     */
+    protected $optionGroups;
+
+    /**
+     * The minimum query length of the attachment action.
+     *
+     * @var int
+     */
+    protected $minQueryLength = 1;
+
+    /**
+     * The data source of the attachment action.
+     *
+     * @var string
+     */
+    protected $dataSource = 'static';
+
+    /**
+     * Set the name of the field.
+     *
+     * @param  string $name
+     *
+     * @return $this
+     */
+    public function name($name)
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * Set the content of the field.
+     *
+     * @param  string $content
+     *
+     * @return $this
+     */
+    public function text($content)
+    {
+        $this->content = $content;
+
+        return $this;
+    }
+
+    /**
+     * Set the type of the field.
+     *
+     * @param  string $type
+     *
+     * @return $this
+     */
+    public function type($type)
+    {
+        $this->type = $type;
+
+        return $this;
+    }
+
+    /**
+     * Set the value of the field.
+     *
+     * @param  string $value
+     *
+     * @return $this
+     */
+    public function value($value)
+    {
+        $this->value = $value;
+
+        return $this;
+    }
+
+    /**
+     * Add a confirmation field for the action.
+     *
+     * @param  \Closure $callback
+     *
+     * @return $this
+     */
+    public function confirmationField(Closure $callback)
+    {
+        $this->confirmationFields[] = $confirmationField = new SlackActionConfirmationField();
+
+        $callback($confirmationField);
+
+        return $this;
+    }
+
+    /**
+     * Set the confirmation fields of the action.
+     *
+     * @param  array $confirmationFields
+     *
+     * @return $this
+     */
+    public function confirmationFields(array $confirmationFields)
```

Comment:
I did this due to several other of the related Slack notification methods type-hinting to array. I can definitely remove that if that's the preference. No huge deal to me.